### ISSUE TITLE
SourcesListLexer selected for any text starting with '#'

### ISSUE THREAD

**Issue by Anteru** [pygments/pygments#599]
(Original issue 892 created by harmic on 2013-07-31T00:18:08.545363+00:00)

The analyse_text method for SourcesListLexer is way too wide:

```
#!python

def analyse_text(text):
        for line in text.split('\n'):
            line = line.strip()
            if not (line.startswith('#') or line.startswith('deb ') or
                    line.startswith('deb-src ') or not line):
                return False
        return True

```

Any line starting with '#' or even an empty line is selected.

```
#!python

>>> from pygments.lexers import guess_lexer
>>> line="#!/usr/bin/perl"
>>> print(guess_lexer(line))
<pygments.lexers.SourcesListLexer>

```

For me the result is that this lexer is chosen for perl files.

Maybe change it to:

```
#!python

if not (line.startswith('deb ') or line.startswith('deb-src '))

```

I notice that it correctly detects python files despite also having a shebang, so maybe this is also related to the order in which the lexers are checked. But I don't know enough about pygments to suggest how that could be changed!

**Comment by Anteru** [comment]
(Original comment by birkenfeld on 2014-10-07T22:20:16.780122+00:00)

Closes #892: fix overzealous analyse_text of the SourcesListLexer.

Also remove any "default" return from analyse_text methods.

→ <<cset cac33c1aeee1>>

**Comment by Anteru** [comment]
(Original comment by birkenfeld on 2014-10-07T22:20:23.974209+00:00)

Thanks for the report!
